src/Shared/ReuseableStringBuilder.cs

```diff
@@ -309,6 +306,9 @@ internal static void Release(StringBuilder returningBuilder)
                 // So the shared builder will be "replaced".
                 if (returningBuilder.Capacity < MaxBuilderSize)
                 {
+                    // ErrorUtilities.VerifyThrow(handouts.TryRemove(returningBuilder, out dummy), "returned but not loaned");
+                    returningBuilder.Clear(); // This is free: it just sets length to zero internally
```

Comment:
If you get a chance before we merge, can you also delete the misleading `//this is free` comment? _I_ was clearly misled by it!